src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs

```diff
@@ -11,11 +12,8 @@ namespace System.Threading
     /// </summary>
     internal sealed partial class PortableThreadPool
     {
-#pragma warning disable IDE1006 // Naming Styles
-        public static readonly PortableThreadPool ThreadPoolInstance = new PortableThreadPool();
-#pragma warning restore IDE1006 // Naming Styles
-
         private const int ThreadPoolThreadTimeoutMs = 20 * 1000; // If you change this make sure to change the timeout times in the tests.
+        private const int SmallStackSizeBytes = 256 * 1024;
```

Comment:
I think the stack size of thread pool threads that run user code should also be configurable (currently they always use the default stack size). It seems like a lot of threads would be necessary to exhaust the address space on 32-bit platforms, the machine would likely run out of physical memory before that. It makes sense to use a smaller stack size by default on 32-bit platforms, perhaps for all threads. In this case the small stack size is used for thread pool threads that don't run user code, it would have to be tested to see if lowering the size would be ok. For now I'm just using the same stack size as the native side and it seems to be sufficient from my testing.